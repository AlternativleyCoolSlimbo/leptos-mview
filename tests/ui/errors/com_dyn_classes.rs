use leptos::*;
use leptos_mview::mview;

#[component]
fn AComponent(
    #[prop(into, default="".into())] class: TextProp,
    #[prop(optional)] id: &'static str,
) -> impl IntoView {
    mview! {
        div class=f["my-class {}", class.get()] {id};
    }
}

fn missing_closure() {
    _ = mview! {
        AComponent class:red=true;
    };
}

fn incorrect_type() {
    _ = mview! {
        AComponent class:red=["not this"];
    };
}

#[component]
fn Nothing() -> impl IntoView {}

// these spans are actually fine, there's a blank info message at `mview!` for
// some reason.

fn no_attribute_reactive() {
    _ = mview! {
        Nothing class:red=[true];
    };
}

fn no_attribute_static() {
    _ = mview! {
        Nothing.red;
    };
}

fn no_attribute_id() {
    _ = mview! {
        Nothing #unique;
    };
}

fn main() {}
