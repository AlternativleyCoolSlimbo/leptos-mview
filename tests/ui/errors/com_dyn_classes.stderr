error[E0618]: expected function, found `bool`
  --> tests/ui/errors/com_dyn_classes.rs:16:30
   |
16 |         AComponent class:red=true;
   |                              ^^^^ call expression requires function

error[E0308]: mismatched types
  --> tests/ui/errors/com_dyn_classes.rs:22:30
   |
22 |         AComponent class:red=["not this"];
   |                              ^^^^^^^^^^^^
   |                              |
   |                              expected `bool`, found `&str`
   |                              arguments to this function are incorrect
   |
note: method defined here
  --> $RUST/core/src/bool.rs
   |
   |     pub fn then_some<T>(self, t: T) -> Option<T> {
   |            ^^^^^^^^^

error[E0599]: no method named `class` found for struct `EmptyPropsBuilder` in the current scope
  --> tests/ui/errors/com_dyn_classes.rs:34:23
   |
33 |       _ = mview! {
   |  _________-
34 | |         Nothing class:red=[true];
   | |                      -^^^ method not found in `EmptyPropsBuilder`
   | |______________________|
   |

error[E0599]: no method named `class` found for struct `EmptyPropsBuilder` in the current scope
  --> tests/ui/errors/com_dyn_classes.rs:40:17
   |
39 |       _ = mview! {
   |  _________-
40 | |         Nothing.red;
   | |                -^^^ method not found in `EmptyPropsBuilder`
   | |________________|
   |

error[E0599]: no method named `id` found for struct `EmptyPropsBuilder` in the current scope
  --> tests/ui/errors/com_dyn_classes.rs:46:18
   |
45 |       _ = mview! {
   |  _________-
46 | |         Nothing #unique;
   | |                 -^^^^^^ method not found in `EmptyPropsBuilder`
   | |_________________|
   |
